src/arduino.cc/builder/ctags/ctags_parser.go

```diff
@@ -229,7 +229,14 @@ func parseTag(row string) *types.CTag {
 	parts := strings.Split(row, "\t")
 
 	tag.FunctionName = parts[0]
-	tag.Filename = parts[1]
+	// This unescapes any backslashes in the filename. These
+	// filenames that ctags outputs originate from the line markers
+	// in the source, as generated by gcc. gcc escapes both
+	// backslashes and double quotes, but ctags ignores any escaping
+	// and just cuts off the filename at the first double quote it
+	// sees. This means any backslashes are still escaped, and need
+	// to be unescape, and any quotes will just break the build.
+	tag.Filename = strings.Replace(parts[1], "\\\\", "\\", -1)
```

Comment:
Can we use `utils.ParseCppString`?

Having a special handling here looks weird, I'd use the full parsing even if ctags doesn't handle `\"`. Or maybe there are corner cases I'm missing?